xbmc/platform/posix/storage/discs/DiscDriveHandlerPosix.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DiscDriveHandlerPosix.h"
+
+#include "storage/cdioSupport.h"
+#include "utils/log.h"
+
+#include <memory>
+
+namespace
+{
+constexpr int MaxOpenRetries = 3;
+}
+using namespace MEDIA_DETECT;
+
+
+std::shared_ptr<IDiscDriveHandler> IDiscDriveHandler::CreateInstance()
+{
+  return std::make_shared<DiscDriveHandlerPosix>();
+}
+
+DriveState DiscDriveHandlerPosix::GetDriveState(const std::string& devicePath)
+{
+  const std::shared_ptr<CLibcdio> libCdio = CLibcdio::GetInstance();
+  if (!libCdio)
+  {
+    CLog::LogF(LOGERROR, "Failed to obtain libcdio handler");
+    return DriveState::DRIVE_NOT_READY;
+  }
+
+  DriveState driveStatus = DriveState::DRIVE_NOT_READY;
```

Comment:
you could move decl of this var to the beginning of this method, init it with DRIVE_NOT_READY and return this var everywhere in this method intstead of sometimes returning DRIVE_NOT_READY directly sometimes the var's value

Reply:
I've come across this one multiple times and I think it comes down to personal preference (howie usually requests the opposite). Personally I think explicitly returning `DRIVE_NOT_READY` on those "exceptions" is clear than defining it on top of the method and just return. What if the code is changed in-between in the future?

Reply:
iirc in some other pr you `return(ed) success; (an enum value)` and that was a bit misleading, thus i requested to return it explicitly. but `return driveStatus` is fine imo. it‘s clear